pyiron_base/storage/parameters.py

```diff
@@ -499,6 +499,14 @@ def define_blocks(self, block_dict):
             raise AssertionError()
         self._block_dict = block_dict
 
+    def to_dict(self):
+        data_dict = self._type_to_dict()
+        data_dict["data_dict"] = self._dataset
+        return data_dict
+
+    def from_dict(self, generic_parameters_dict):
+        self._dataset = generic_parameters_dict
```

Comment:
```suggestion
        self._dataset = generic_parameters_dict["data_dict"]
```
imo `to_dict` and `from_dict` should be symmetric, meaning I should be able to call `obj.from_dict(obj.to_dict()) == obj`.  Everything else outside should treat the dict as opaque.

Reply:
I made the change in a separate commit, to also update the `from_hdf()` function accordingly.